frontend/src/component/admin/roles/RoleForm/RolePermissionCategories/RolePermissionProject.tsx

```diff
@@ -0,0 +1,148 @@
+import { useMemo } from 'react';
+import {
+    Box,
+    Checkbox,
+    FormControlLabel,
+    styled,
+    Typography,
+} from '@mui/material';
+import type { ICheckedPermissions, IPermission } from 'interfaces/permissions';
+import { getRoleKey } from 'utils/permissions';
+import {
+    type PermissionCategory,
+    PROJECT_PERMISSIONS_STRUCTURE,
+} from '@server/types/permissions';
+
+interface IEnvironmentPermissionAccordionProps {
+    permissions: IPermission[];
+    checkedPermissions: ICheckedPermissions;
+    onPermissionChange: (permission: IPermission) => void;
+}
+
+const StyledGrid = styled('div')(({ theme }) => ({
+    display: 'grid',
+    gridTemplateColumns: '1fr 1fr',
+    gap: theme.spacing(1),
+    [theme.breakpoints.down('md')]: {
+        gridTemplateColumns: '1fr',
+    },
+}));
+
+const StyledLabel = styled(Typography)(({ theme }) => ({
+    lineHeight: 1.2,
+    marginBottom: theme.spacing(1),
+}));
+
+const StyledSectionTitle = styled(Typography)(({ theme }) => ({
+    fontWeight: theme.typography.fontWeightBold,
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(1),
+}));
+
+export const ProjectRolePermissionCategory = ({
+    permissions,
+    checkedPermissions,
+    onPermissionChange,
+}: IEnvironmentPermissionAccordionProps) => {
+    const allPermissions = useMemo(
+        () =>
+            permissions?.map((permission) => getRoleKey(permission)).sort() ||
+            [],
+        [permissions],
+    );
+
+    const otherPermissions = useMemo(() => {
+        const allStructuredPermissions = PROJECT_PERMISSIONS_STRUCTURE.flatMap(
+            (category) =>
+                category.permissions.map(([permission]) => permission),
+        ).sort() as string[];
+
+        return allPermissions.filter(
+            (permission) => !allStructuredPermissions.includes(permission),
+        );
+    }, [permissions]);
+
+    const permissionsStructure = useMemo(
+        () =>
+            [
+                ...PROJECT_PERMISSIONS_STRUCTURE,
+                {
+                    label: 'Other',
+                    permissions: otherPermissions.map((p) => [p]),
+                } as PermissionCategory,
```

Comment:
If there is a discrepancy between hard-coded frontend structure, and that's received from API, this makes sure all backend items are shown.